pennylane/devices/tests/test_measurements.py

```diff
@@ -1783,15 +1783,15 @@ def circuit():
             qml.X(0)
             return MyMeasurement()
 
-        if isinstance(dev, qml.Device):
-            with pytest.warns(
+        with (
+            pytest.warns(
                 UserWarning,
-                match="Requested measurement MyMeasurement with finite shots",
-            ):
-                circuit()
-        else:
-            with pytest.raises(qml.DeviceError):
-                circuit()
+                match="MyMeasurement with finite shots; the returned state information is analytic",
+            )
+            if isinstance(dev, qml.devices.LegacyDevice)
```

Comment:
Because the facade is merely an adapter. The underlying code for executing measurements is still different and will provide different warnings/errors.